### ISSUE TITLE
DataTable accessibility breaks when size prop is defined

### ISSUE THREAD

**Issue by srallen** [grommet/grommet#3006]
### Expected Behavior
VoiceOver should be able to read out all of the columns and rows of the table.

### Actual Behavior
When the size prop on the DataTable component is defined, VoiceOver no longer reads out the rows:

<img width="658" alt="Screen Shot 2019-04-08 at 3 39 31 PM" src="https://user-images.githubusercontent.com/5016731/55732826-8e9cd780-5a14-11e9-9697-51678adc24f3.png">

You can see this happening in the Grommet storybook:

https://storybook.grommet.io/?path=/story/datatable--sized-datatable

The sized datatable and the controlled datatable examples have size defined and do not get read out correctly. The others do get read out with the correct number of rows. 

### Steps to Reproduce

1. Set the size prop on a DataTable
2. Try to use VoiceOver
3. Navigate to the table using VoiceOver by hitting control + option + command + T

### Your Environment

- Grommet version: I'm using 2.5.5, but it's in the storybook, so I presume a bug in the latest version too
- Browser Name and version: Chrome 73
- Operating System and version (desktop or mobile): OS X 10.14.4

**Comment by mrebb** [comment]
It looks like it is possible to have either `size` or fix accessibility issue on DataTable. Main reason behind `voice over` not detecting the number of rows or columns is, based on how table headers and body style is updated when `size` prop is provided in current logic (StyledDataTable.js). If we fix the table elements to make it compatible for accessibility then we mostly likely lose the main purpose of `size` prop as setting height on table body is not possible without applying `display` as `block` on table body . Solution is possible when setting height on `table-row-group` is supported by w3c. See this [issue](https://github.com/w3c/csswg-drafts/issues/476) for reference. 

Any other thoughts for possible fix?

**Comment by mrebb** [comment]
@srallen Could you please share your thoughts on this?

**Comment by srallen** [comment]
My team is required to make a best effort to meet US Section 508 accessibility requirements for the web. Because the size prop in its current implementation breaks accessibility by screen readers of the table, then the choice is made for us, we can't use it. However, I'm not comfortable with telling Grommet to remove it or change it entirely as others who may use Grommet don't have to meet as strict of guidelines as we do. I'd be for having a warning in the docs about usage of this prop as an immediate thing that can be done. For longer term, what we did is set the size of the outer `Box` and set it to overflow. Not ideal, I know, but like I said, we have to have an accessible site.

**Comment by jcfilben** [comment]
Closing since voice over is now working with DataTable when the `size` prop is defined.
